### ISSUE TITLE
[Bug] No connection detected

### ISSUE THREAD

**Issue by etiennebacher** [ProtonMail/proton-mail-android#59]
**Describe the bug**
When opening the app, or opening a mail, or writing a mail, an red error message appears at the bottom of the screen. The message says: "No connection detected (troubleshooting)" (this is an approximate translation, the original message in French says: "Aucune connexion détectée (dépannage)").

**To Reproduce**
1. Open the app, or open a mail, or click on "Write"
2. See error

**Expected behavior**
No message should be displayed as I have a constant Internet connection.

**Screenshots**
![dest](https://user-images.githubusercontent.com/52219252/103134156-0a3a2e80-46a7-11eb-8762-5e3421edf260.jpg)

**Smartphone (please complete the following information):**
 - Device: Huawei Honor 8
 - OS: Android 7.0
 - Version: 1.13.22

**Additional context**
/

**Comment by marinomeneghel** [comment]
Hi @etiennebacher , thanks for reporting this.
This issue is known and we're taking actions to improve such behaviour. A fix is currently being rolled out on the latest beta (v. 751) please feel free to try it out and let us know if that solves it for you!

**Comment by etiennebacher** [comment]
Hi, if I'm not mistaken, v. 751 was deployed (Aurora Store tells me it is v.1.13.24.751), but the bug is still present on my phone.

**Comment by NitroNils** [comment]
I have 1.13.25 and the bug is still there.

**Comment by marinomeneghel** [comment]
Thanks for your feedback, some additional improvements are being worked based on it.

Is there any additional info you can provide in regards to when this happens? (eg. Always when on data connection, poor signal...)

**Comment by etiennebacher** [comment]
This red popup always appears, whatever the signal quality or the type of connection (4G, WiFi...). 

The time during which the popup stays is variable but I don't know what influences it. Sometimes, when the signal is good, the popup appears and disappears very quickly, but sometimes it also stays there indefinitely if I don't touch anything else.

Also, it appears every time I run the app, whether ProtonMail was already running in the background or not.

I hope this will help you fix this.

**Comment by Ammako** [comment]
I tried to send a mail. Stuck for 5 minutes in Drafts, failing to send. App keeps coming up with "No connection detected" even though in not disconnected.

Takes forever for the content of an email to load, so I can read it and reply. The unsendable draft finally disappears, I rewrite my email and try to send it again. I end up with 6 drafts in my Drafts folder, half of which are empty, the rest are all copies of the email I just tried to send.

Deleted them all, tried again a third time to send. Stuck in Drafts on Sending again. Keeps complaining about no connection every so often.

Out of the blue, suddenly I get Message Sent notification. The email briefly shows in the Sent folder, but immediately disappears. If I open All Mail folder, I can see it in there and it claims that it's in the Sent folder. But it doesn't show up there.

I move it to Archive from All Mail, it never actually shows up in Archive. Meanwhile I have no idea if I've actually correctly sent my email or not, or maybe my recipient just received the same emaol 15 times. Who knows.

This app is... not great.

**Comment by marinomeneghel** [comment]
Thanks both for the additional details, that's of great help. 

We're actively taking actions to improve the issues with connectivity banner which are in our knowledge, we'll provide updates back here once such improvements start being rolled out. 

Meanwhile please feel free to add any additional feedback or other cases you encounter that you feel weren't raised already, so that we can consider them.

**Comment by Ammako** [comment]
I'd say 95% of the time it works fine. I've not encountered this issue more than maybe two or three times in a year.

It would be pretty cool if it could be fixed, but personally I feel that efforts would be better used towards the v4 beta that's meant to replace the current app.

Then again, that's just me. The app still works 95% of the time and I can live with it until the new app is available, but if others are getting this error a lot more often and it prevents them from using the app at all, then yeah, it's definitely worth looking into it.

**Comment by etiennebacher** [comment]
@Ammako I think your problem is a bit different than mine: I don't have any problem to send email, I just have a banner that tells me (all the time) that I have no connection whereas I have.

**Comment by Ammako** [comment]
Hm 🤔 so I guess it might simply be a coincidence that the "No connection detected" happened at the same time.

Anyway, https://github.com/ProtonMail/proton-mail-android/issues/62

**Comment by etiennebacher** [comment]
Hello, small update on this issue: I updated ProtonMail to 1.13.29.766 and the issue is now a bit different.

Now, the error message has an orange background (and not red anymore), and the message is "You are offline". The message appears and disappears very quickly so I can't make a screenshot to show you. It still appears everytime I open the app, whether it was already running in background or not, and whatever the network I use.

**Comment by eddex** [comment]
I have the same issue.
- When connected to a WiFi network, the app works fine.
- When I use a mobile connection, the app shows a red error message at the bottom "Server not reachable...".

I get notifications about new emails but they are not displayed in the app.

App version: 1.13.29.766
Phone: OnePlus 2
Android version: 6.0.1 (Oxygen OS)

**Comment by etiennebacher** [comment]
@eddex I don't think this is the same issue since the error I'm talking about also appears on a wifi network, and the error message is not the same

**Comment by etiennebacher** [comment]
> Now, the error message has an orange background (and not red anymore), and the message is "You are offline". The message appears and disappears very quickly so I can't make a screenshot to show you. It still appears everytime I open the app, whether it was already running in background or not, and whatever the network I use.

Hello, just want to inform that I still have this bug in version 1.13.38

**Comment by suj490** [comment]
I am also having problems with ProtonMail. Each time I send an email I get the following message (see attached). Getting ready to close my ProtonMail account if this keeps happening because emails that need to be responded to in a timely manner are not going thru until the next day when I open the app for the first time. 
![Uploading 53632A60-6D35-4B59-8884-B0A70373EB2A.png…]()
